content/blog/using-apache-spark-dataframes-for-processing-of-tabular-data.md

```diff
@@ -0,0 +1,362 @@
+
+**The sample data sets**
+
+We will use two example datasets - one from <a target='\_blank'  href='http://www.modelingonlineauctions.com/datasets'>eBay online auctions</a> and one from the <a target='\_blank'  href='https://data.sfgov.org/Public-Safety/SFPD-Incidents-from-1-January-2003/tmnf-yvry'>SFPD Crime Incident Reporting system</a>.
+
+The eBay online auction dataset has the following data fields:
+
+***auctionid*** - unique identifier of an auction\
+***bid*** - the proxy bid placed by a bidder\
+***bidtime*** - the time (in days) that the bid was placed, from the start of the auction\
+***bidder*** - eBay username of the bidder\
+***bidderrate*** - eBay feedback rating of the bidder\
+***openbid*** - the opening bid set by the seller\
+***price*** - the closing price that the item sold for (equivalent to the second highest bid + an increment)
+
+The table below shows the data fields with some sample data:
+
+![](/img/blog_sparkdataframes_table1.png)
+
+Using Spark DataFrames we will explore the data with questions like:
+
+* How many auctions were held?
+* How many bids were made per item?
+* What's the minimum, maximum, and average number of bids per item?
+* Show the bids with price > 100
+
+The table below shows the SFPD data fields with some sample data:
+
+![](/img/blog_sparkdataframes_table2.png)
+
+Using Spark DataFrames, we will explore the SFPD data with questions like:
+
+* What are the top 10 Resolutions?
+* How many Categories are there?
+* What are the top 10 incident Categories?
+
+## Loading data into Spark DataFrames
+
+Log into the MapR Sandbox, as explained in [Getting Started with Spark on MapR Sandbox](https://developer.hpe.com/blog/getting-started-with-spark-on-mapr-sandbox/), using userid user01, password mapr. Copy the sample data files to your sandbox home directory /user/user01 using scp. Start the spark shell with:\
+`$ spark-shell`
+
+First, we will import some packages and instantiate a sqlContext, which is the entry point for working with structured data (rows and columns) in Spark and allows the creation of DataFrame objects.\
+(In the code boxes, <font color="green">comments are in Green</font> and <font color="#005CB9">output is in Blue</font>)
+
+<pre>
+<font color="green">//  SQLContext entry point for working with structured data</font>
+val sqlContext = new org.apache.spark.sql.SQLContext(sc)
+<font color="green">// this is used to implicitly convert an RDD to a DataFrame.</font>
+import sqlContext.implicits._
+<font color="green">// Import Spark SQL data types and Row.</font>
+import org.apache.spark.sql._
+</pre>
+
+Below we load the data from the ebay.csv file into a Resilient Distributed Dataset (RDD). RDDs can have transformations and actions; the first() action returns the first element in the RDD, which is the String `“8213034705,95,2.927373,jake7870,0,95,117.5,xbox,3”`
```

Comment:
Add a comma after "Below". Also, should some of this be in a code block?
![image](https://user-images.githubusercontent.com/77496261/137518468-3df79603-8eb8-4302-ab29-d91af9092e3d.png)

Reply:
done